### PR TITLE
fix: skip const-fold for `arange` for attention mask

### PR DESCRIPTION
**Pull request by fs-eire**
# Description

This change introduced a const-fold exclusion rule "attention_mask_arange". With this rule, generated causal mask will be preserved as operations so that later TRT/Myelin can correctly generate optimized layer.

#4450 is required as prerequisite for this PR.

Fixes # (issue)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: tests/py/dynamo/lowering/test_attention_mask_constant_fold_exclusion.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            all(
+                node.meta.get(CONSTANT_FOLD_EXCLUSION_META_KEY, False)
+                for node in arange_nodes
+            )
+        )
+
+    def test_decomposed_attention_mask_aranges_are_not_folded(self):
+        self._assert_only_attention_aranges_survive(decompose_attention=True)
+
+    def test_ia_attention_mask_aranges_are_not_folded(self):
```
Comment (micwill755) [inline tests/py/dynamo/lowering/test_attention_mask_constant_fold_exclusion.py:73]: What is "ia"? Is this a typo?

File: py/torch_tensorrt/dynamo/lowering/passes/mark_constant_fold_exclusions.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from typing import Any, Optional
+
+import torch
+from torch_tensorrt.dynamo.lowering.constant_fold_exclusions._core import (
```
Comment (micwill755) [inline py/torch_tensorrt/dynamo/lowering/passes/mark_constant_fold_exclusions.py:4]: Shouldnt we eagerly import attention_mask here, or via the package __init__, so the rule is always registered when this pass runs?

**Review by zewenli98 — COMMENTED** [review]
It looks this PR overlaps with #4450. Can you merge as one PR and close the other for better review?

**Comment by fs-eire** [comment]
> It looks this PR overlaps with #4450. Can you merge as one PR and close the other for better review?

There are 2 commits in this PR, and the first commit is exactly the content of #4450, so you can choose to diff only the second commit to see the change.

I will merge #4450 first and rebase this PR then.
